### PR TITLE
Make responses more explicit.

### PR DESCRIPTION
**Pull request by EBashkoff**
Please review code that more explicitly provides response codes to the caller.

**Thank you!!**

### CODE REVIEW

File: lib/contactually.rb [hunk 1]
```diff
@@ -43,7 +45,7 @@ def param_fields(params)
     end
 
     def make_call(composite, params = {})
-      return false unless get_methods(composite)
+      return ReturnValue.new(400, "Bad action/method") unless get_methods(composite)
```
Comment (mlpinit) [inline lib/contactually.rb]: Can we return what the bad action method was in the response as well?

File: lib/contactually.rb [hunk 2]
```diff
@@ -52,7 +54,16 @@ def make_call(composite, params = {})
         curl.headers['Accept']       = 'application/json'
         curl.headers['Content-Type'] = 'application/json'
       end
-      JSON.load("[#{response.body_str}]").first
+
+      if [200, 201, 202].include?(response.response_code)
+        return ReturnValue.new(
+          response.response_code,
+          JSON.load("[#{response.body_str}]"
```
Comment (mlpinit) [inline lib/contactually.rb]: I think line 61 and 62 are more readable if they are on the same line.

**Comment by mlpinit** [comment]
@EBashkoff We should try to return the same type(Array in this case) as a response for the second argument inside the ReturnValue struct.

**Comment by guillec** [comment]
@EBashkoff @mlpinit can you give me a background on this PR?
I think I understand why you did what you did but would love a little explanation.

Thanks!

**Comment by guillec** [comment]
@EBashkoff @mlpinit please check.

**Comment by guillec** [comment]
@EBashkoff @mlpinit please take a look at what I did, I like the idea of the ReturnValue.
I just did some cleaning up and I agree that we should always return the same type in the response_body.

**Comment by guillec** [comment]
I didnt get a change to manually test :( so if you could do a quick check then merge and I can then create the new gem.

**Pull request merged**